src/components/MultiHopTrade/hooks/useHopHelper.tsx

```diff
@@ -0,0 +1,134 @@
+import { useMemo } from 'react'
+import { useAccountIds } from 'components/MultiHopTrade/hooks/useAccountIds'
+import { useGetTradeQuotes } from 'components/MultiHopTrade/hooks/useGetTradeQuotes'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit } from 'lib/math'
+import { SwapperName } from 'lib/swapper/api'
+import { selectFeeAssetById } from 'state/slices/assetsSlice/selectors'
+import {
+  selectPortfolioCryptoBalanceBaseUnitByFilter,
+  selectPortfolioCryptoPrecisionBalanceByFilter,
+} from 'state/slices/common-selectors'
+import { useAppSelector } from 'state/store'
+
+export const useHopHelper = () => {
+  const { selectedQuote } = useGetTradeQuotes()
+  const quoteData = useMemo(
+    () => (selectedQuote?.data?.isOk() ? selectedQuote.data.unwrap() : undefined),
+    [selectedQuote?.data],
+  )
+
+  const firstHop = useMemo(() => quoteData?.steps[0], [quoteData?.steps])
+  const lastHop = useMemo(() => quoteData?.steps[quoteData.steps.length - 1], [quoteData?.steps])
+
+  const firstHopSellAsset = useMemo(() => firstHop?.sellAsset, [firstHop?.sellAsset])
+  const firstHopBuyAsset = useMemo(() => firstHop?.buyAsset, [firstHop?.buyAsset])
+  const lastHopSellAsset = useMemo(() => lastHop?.sellAsset, [lastHop?.sellAsset])
+  const lastHopBuyAsset = useMemo(() => lastHop?.buyAsset, [lastHop?.buyAsset])
+
+  const { sellAssetAccountId: firstHopSellAssetAccountId } = useAccountIds({
+    buyAsset: firstHopBuyAsset,
+    sellAsset: firstHopSellAsset,
+  })
+
+  const { sellAssetAccountId: lastHopSellAssetAccountId } = useAccountIds({
+    buyAsset: lastHopBuyAsset,
+    sellAsset: lastHopSellAsset,
+  })
```

Comment:
all buy/sell assets are guaranteed to exist so doesnt need to be optional

Reply:
As discussed, leaving these guys as optional for now.